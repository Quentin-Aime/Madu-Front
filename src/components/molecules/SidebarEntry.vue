<template>
  <div class="entry" :class="{selected: isSelected}" @click="redirect">
    <el-tooltip
      class="item"
      effect="dark"
      :content="tooltip"
      placement="right-start"
    >
      <img class="icon" :src="imgSrc" />
    </el-tooltip>
  </div>
</template>

<script>
export default {
  props: {
    linkName: {
      type: String,
      required: true
    },
    imageName: {
      type: String,
      required: true
    },
    tooltip: {
      type: String,
      default: "Home"
    }
  },
  data: function() {
    return {
    };
  },
  computed: {
    isSelected() {
      console.debug(this.$route); //eslint-disable-line
      return this.$route.name === this.linkName;
    },
    imgSrc() {
      return require(`../../static/img/${this.imageName}_${this.isSelected ? 'white' : 'blue'}.png`);
    }
  },
  mounted() {
  },
  methods: {
    redirect() {
      this.$router.push({
        name: this.linkName
      });
    }
  }
};
</script>

<style lang="scss">
.entry {
  display: flex;
  align-items: center;
  width: 100%;
  color: #c0c5d2;
  height: 56px;
  + .entry {
    margin-top: 25px;
  }
  .icon {
    margin: auto;
  }
  &.selected {
    border-right: 3px solid white;
  }
  &:hover {
    cursor: pointer;
    background-color: #5B74ED;
  }
}
</style>
