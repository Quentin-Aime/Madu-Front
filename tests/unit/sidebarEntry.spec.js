import { shallow } from "@vue/test-utils";
import SidebarEntry from "../../src/components/molecules/sidebarEntry.vue";

it("should match snapshot", () => {
  const wrapper = shallow(SidebarEntry);

  expect(wrapper.element).toBeDefined();
  expect(wrapper.element).toMatchSnapshot();
});
