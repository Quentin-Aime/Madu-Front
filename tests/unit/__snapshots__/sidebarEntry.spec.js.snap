// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  class="entry"
>
  <el-tooltip
    class="item"
    content="Home"
    effect="dark"
    placement="right-start"
  >
    <i
      style="font-size: 2rem;"
    />
  </el-tooltip>
</div>
`;
